### PR TITLE
Clean up lightbox docs and runtime validation

### PR DESCRIPTION
**Pull request by cathyxz**
Runtime validation and docs were cleaned up to reflect the following: 
1. We decided to drop support for `<amp-instagram>`, `<amp-facebook>` pending layout intrinsic or an implementation for `object-fit: contain` for layout fill. 
2. We decided to drop support for `<amp-anim>` due to performance issues associated with gifs, and encourage the use of `<amp-video>` with no controls and autoplay instead.

### CODE REVIEW

File: extensions/amp-lightbox-gallery/validator-amp-lightbox-gallery.protoascii [hunk 1]
```diff
@@ -35,8 +35,5 @@ attr_lists: {
   attrs: {
     name: "lightbox-thumbnail-id"
     value_regex_casei: "^[a-z][a-z\\d_-]*"
-    trigger: {
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/validator-amp-lightbox-gallery.protoascii]: /cc @Gregable 
This makes the validation much more permissive than it should be, but allows all legitimate use cases to work. Everything that should be disallowed is currently covered by runtime validation instead, in the form of `user().assert` logs. I can revert now / revisit when we have a good approach to validate this.

File: extensions/amp-lightbox-gallery/amp-lightbox-gallery.md [hunk 2]
```diff
@@ -116,16 +116,14 @@ Lightboxed items have a thumbnail gallery view. You can optionally specify a thu
 #### Example: using `lightbox-thumbnail-id` to specify a thumbnail
```
Comment (cathyxz) [inline extensions/amp-lightbox-gallery/amp-lightbox-gallery.md]: /cc @bpaduch replaced the `<amp-facebook>` with `<amp-youtube>` instead. There's a tested version of this that works in the `test/manual/amp-lightbox-gallery-launch.amp.html`.

File: test/manual/amp-lightbox-gallery-launch.amp.html [hunk 3]
```diff
@@ -179,7 +171,8 @@ <h2>AMP-YOUTUBE Layout Test</h2>
     <amp-youtube width="480"
       height="270"
       layout="responsive"
-      data-videoid="lBTCB7yLs8Y">
+      data-videoid="lBTCB7yLs8Y"
+      lightbox-thumbnail-id="fb-thumbnail-img">
```
Comment (cvializ) [inline test/manual/amp-lightbox-gallery-launch.amp.html]: should this be `yt-thumbnail-img`?

File: extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js [hunk 4]
```diff
@@ -18,7 +18,6 @@ import {AmpEvents} from '../../../../src/amp-events';
 import {CommonSignals} from '../../../../src/common-signals';
 import {
   LIGHTBOX_THUMBNAIL_AD,
-  LIGHTBOX_THUMBNAIL_GIF,
```
Comment (cvializ) [inline extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js]: Is this value still used anywhere in the code? Can it be removed from `lightbox-placeholders.js`?

File: test/manual/amp-lightbox-gallery-launch.amp.html (hunk 3 above)
Comment (cathyxz) [reply test/manual/amp-lightbox-gallery-launch.amp.html]: Yup! Thanks for the catch!

File: extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js (hunk 4 above)
Comment (cathyxz) [reply extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js]: I was debating this, but I think yes.

File: test/manual/amp-lightbox-gallery-launch.amp.html (hunk 3 above)
Comment (cathyxz) [reply test/manual/amp-lightbox-gallery-launch.amp.html]: Oh actually this doesn't matter as it is test code. But might as well change it to proper naming anyway.

**Pull request merged**